QtCollider/CMakeLists.txt

```diff
@@ -134,8 +144,18 @@ if(QT_COLLIDER_LANG_CLIENT)
     list(APPEND QT_COLLIDER_SRCS ${QT_COLLIDER_DIR}/LanguageClient.cpp)
 endif()
 
+if(QT_WEBENGINE)
+  list(APPEND QT_COLLIDER_HDRS
+    ${QT_COLLIDER_DIR}/widgets/QcWebView.h
+    ${QT_COLLIDER_DIR}/widgets/web_page.hpp
+  )
+  list(APPEND QT_COLLIDER_SRCS
+    ${QT_COLLIDER_DIR}/widgets/QcWebView.cpp
+    ${QT_COLLIDER_DIR}/widgets/web_page.cpp
+  )
+endif()
```

Comment:
A better way to do this is to wrap the entire file in `#ifdef QT_WEBENGINE`. By excluding them here you're also excluding them from any IDE project generated by CMake. Someone may want to browse it in the project even if it's not part of the build per se. You also save yourself from having to put #ifdef guards around the #include statements.

Reply:
okay uhhhhh even if `QT_WEBENGINE` is on, the `#ifdef QT_WEBENGINE` *still* produces the warning. so there's no way around this :/

Reply:
the warning is because we're calling qt_wrap_cpp on it later; it is still possible to avoid the warning

Reply:
you could do it this way - assuming the problem is really just that we're handing `moc` empty header files. create a new list, `QT_COLLIDER_WEBENGINE_HDRS`, with those problematic files, that you conditionally pass into `qt_wrap_cpp` (line 160 below). then pass that list as another argument to the `list(APPEND` call on line 178.

Reply:
nah man -- i did resolve the conversation, but that was before you left your comment on how to work around the qt warning, and i hadn't touched the pr since then. its just the way github shows stuff can cause confusion.

thanks for the explanation, i'll give it a shot.

Reply:
ok, no worries!

Reply:
i was able to make the cmake change suggested that ensures that the webview headers are always in `QT_COLLIDER_SRCS` unconditionally. however i simply can't get this `#ifdef` to work. when `SC_USE_WEBENGINE` is off, it's fine. but when `SC_USE_WEBENGINE` is on, the `moc` warning is still triggered, and furthermore the whole build fails with a linker error. if you would like to try it, i have it up on this branch:
 https://github.com/snappizz/supercollider/commit/bea912994c337680aa25bafc885dae53fd229c06

there aren't any actual explicit `#include "QcWebView.h"`'s anywhere except in `QcWebView.cpp`, so there is really no need at all for this `#ifdef`.

Reply:
I can't recreate what you're saying locally with your branch, so I threw it into CI: https://travis-ci.org/supercollider/supercollider/builds/427962590

Reply:
@snappizz oops, I missed where you said this happens with the option ON.